backend/init_data/skills/dingtalk-knowledge/provider.py

```
@@ -0,0 +1,650 @@
+            temp_dir = f"/tmp/dingtalk_{int(time.time())}"
+            save_path = f"{temp_dir}/{filename}"
+
+            # Create temp directory
+            await sandbox.files.make_dir(temp_dir)
+
+            # Write content to file using Python in sandbox
+            # Escape content for shell command
+            escaped_content = content.replace("'", "'\"'\"'")
+            python_cmd = f'python3 -c \'open("{save_path}", "w", encoding="utf-8").write("{escaped_content}")\''
+
+            logger.info(f"[SaveDingTalkContentTool] Writing content to file")
```

Ruff (0.15.10)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)